### DISCUSSION TITLE
Improve Indexing Performance for NTFS Disks with Master File Table (MFT)

### DISCUSSION THREAD

**Discussion by greyovo** [spacedriveapp/spacedrive#2714] [Feature Request]
For NTFS disks, consider using Master File Table (MFT) to greatly optimize the indexing speed (a full disk index can be established in about 10 seconds) instead of scanning through IO. MFT is a file index table on the NTFS disk system. By reading it, you can quickly obtain the index of the entire disk, such as the well-known software [Everything](https://www.voidtools.com), [Listray](https://www.listary.com) does just that.

**Comment by greyovo** [comment]
FYI: https://learn.microsoft.com/en-us/windows/win32/fileio/master-file-table

**Reply by matheus-consoli** [reply]
Thanks for pointing that out @greyovo, we'll take a look into it. Anything that makes the indexer faster makes me happy :smile:
